code/modules/projectiles/updated_projectiles/guns/revolvers.dm

```diff
@@ -444,3 +444,45 @@
 	burst_delay = 0.5 SECONDS
 	scatter_unwielded = 20
 	damage_mult = 1.05
+//-------------------------------------------------------
+//.357 AUTOREVOLVER//Based on the Manurhin MR73 autorevolver carried by GIGN operators.
+obj/item/weapon/gun/revolver/mr76
+	name = "\improper Manurhin MR76 autorevolver"
+	desc = "A lean .357 made by Manurhin. One of the most reliable revolvers you will ever be able to use."
+	icon_state = "mr76"
+	item_state = "mr76"
+	caliber = ".357" //codex
+	max_shells = 6 //codex
+	fire_sound = 'sound/weapons/guns/fire/revolver.ogg'
+	current_mag = /obj/item/ammo_magazine/internal/revolver/small
+	force = 6
+	attachable_allowed = list(
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/flashlight,
+						/obj/item/attachable/extended_barrel,
+						/obj/item/attachable/compensator,
+						/obj/item/attachable/scope,
+						/obj/item/attachable/lasersight,
+						/obj/item/attachable/scope/mini)
+	attachable_offset = list("muzzle_x" = 32, "muzzle_y" = 22,"rail_x" = 13, "rail_y" = 24, "under_x" = 25, "under_y" = 17, "stock_x" = 25, "stock_y" = 17)
+
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO, GUN_FIREMODE_AUTOMATIC)
```

Comment:
theres no such thing as an automatic revolver

Reply:
Double Action revolvers pull the hammer back and push it forward with 1 trigger pull. The full auto mode would just be rapid trigger pulls from the user and QoL so you don't have to get carpel tunnel in order to ahoot quickly.

Reply:
*cough* https://en.wikipedia.org/wiki/Webley%E2%80%93Fosbery_Automatic_Revolver *cough*

Reply:
Oof